src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -564,35 +570,53 @@ internal IList<FileLocation> CreateResponseFilesList(IDictionary<string, string>
 
                 foreach (string key in responseFileToContentsDictionary.Keys)
                 {
-                    var fileLocation = new FileLocation
+                    // If the response file is mentioned in Run.Files, use the FileLocation
+                    // object from there (which, conveniently, already has the FileIndex property
+                    // set); otherwise create a new FileLocation.
+                    FileLocation fileLocation = null;
+                    FileData responseFile = null;
+                    bool existsInRunFiles = _v1FileKeytoV2IndexMap.TryGetValue(key, out int responseFileIndex);
+                    if (existsInRunFiles)
                     {
-                        Uri = new Uri(key, UriKind.RelativeOrAbsolute)
-                    };
-                    fileLocations.Add(fileLocation);
-
-                    if (_currentRun != null && !string.IsNullOrWhiteSpace(responseFileToContentsDictionary[key]))
+                        responseFile = _currentRun.Files[responseFileIndex];
+                        fileLocation = responseFile.FileLocation;
+                    }
+                    else
                     {
-                        // We have contents, so mention this file in _currentRun.files
-                        if (_currentRun.Files == null)
+                        fileLocation = new FileLocation
                         {
-                            _currentRun.Files = new List<FileData>();
-                        }
+                            Uri = new Uri(key, UriKind.RelativeOrAbsolute)
+                        };
+                    }
 
-#if FILES_ARRAY_WORKS
-                        if (!_currentRun.Files.ContainsKey(key))
+                    // If this response file has contents, add it to Run.Files, if it
+                    // isn't already there.
+                    if (!string.IsNullOrWhiteSpace(responseFileToContentsDictionary[key]))
```

Comment:
>(responseFileToContentsDictionary[key] [](start = 50, length = 38)

you retrieve this content twice. why not cache, then set it later? see line 615

Reply:
Fixed.

---
In reply to: [248877992](https://github.com/Microsoft/sarif-sdk/pull/1221#discussion_r248877992) [](ancestors = 248877992)